pkg/util/json/encoded.go

```diff
@@ -436,6 +436,8 @@ func (j *jsonEncoded) FetchValKey(key string) (JSON, error) {
 			}
 			return string(data) >= key
 		})
+		// TODO(ssd): This err will always be nil. It looks
+		// like we intended to assign to it at 433 above.
```

Comment:
Looks like this was added in https://github.com/cockroachdb/cockroach/pull/21676 and I agree it looks like the function passed to `sort.Search` is suppose to set the outer `err` variable.

Reply:
I've pushed a fix that collects the last non-nil error.  I'm mildly nervous since from what I can see the unit test suite doesn't hit many of the error paths in this code.  But I'm made less nervous by the fact that (I believe) an error here would imply that the caller has some incorrectly encoded json and this code already assumes a correct encoding in a number of places.

Reply:
LGTM